controllers/cluster_controller.go

```diff
@@ -199,8 +205,10 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		if len(errs) > 0 {
 			return ctrl.Result{}, kerrors.NewAggregate(errs)
 		}
+	}
 
-		// Requeue so we can check the next time to see if there are still any children left.
+	if !descendants.isEmpty() {
```

Comment:
should we log here to find out which descendant has not been deleted yet?

Reply:
I added some logging - PTAL!